### PR TITLE
initial commit ngrok integration

### PR DESCRIPTION
**Pull request by mkarnowski**
initial ngrok integration commit

### What does this PR do?

Initial commit for ngrok integration tile.  This is a tile-only-listing

### Motivation

ngrok and Datadog partnership

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by drichards-87** [comment]
Created DOCS-5569 for the Docs Team review.

File: ngrok/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+## Overview
+
+ngrok delivers instant ingress to your applications in any cloud, private network, or devices with authentication, load balancing, and other critical controls via our global points of presence.
+
+The ngrok platform includes a Datadog event destination integration.  When used in conjunction with ngrok HTTP events the visualization of valuable application insights is made possible.  Using the the Datadog HTTPS logging endpoint, the integration can quickly be setup through the [ngrok dashboard UI][2].
```
Comment (emarsha94) [inline ngrok/README.md:5]: Can you include the following in the overview? 

1. What does ngrok do in 1-2 sentence. (This was answered in the first paragraph)
2. What will customers get from this integration? (In term of value) This should be clear enough for a new customer to understand what this integration does. 
3. What types of data?

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Thanks for submitting your PR! I left some comments and suggested changes. Additionally, please update your setup instructions to include oauth

File: ngrok/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+This integration includes:
+
+- Dashboard - *ngrok HTTP Events*
```
[the remark this answers is not in the document]
Comment (mkarnowski) [reply ngrok/README.md]: Done

File: ngrok/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- object.conn.client_ip
+- object.conn.server_name
+- object.http.response.status_code
+- object.http.request.method
+- object.http.request.url.path
+- object.conn.server_name
+
+## Troubleshooting
+
+Need help? Contact [Datadog Support][1] or [ngrok Docs][6].
```
[the remark this answers is not in the document]
Comment (mkarnowski) [reply ngrok/README.md]: Fixed

File: ngrok/manifest.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "overview": "README.md#Overview",
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Visualize valuable application insights with ngrok HTTP events.",
+    "title": "ngrok",
+    "media": [
+      {
+        "media_type": "image",
+        "caption": "ngrok HTTP request events overview dashboard.",
```
[the remark this answers is not in the document]
Comment (mkarnowski) [reply ngrok/manifest.json]: done

File: ngrok/manifest.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "image_url": "images/diag2.png"
+      }
+    ],
+    "classifier_tags": [
+      "Category::Developer Tools",
+      "Category::Cloud",
+      "Supported OS::Linux",
+      "Supported OS::Windows",
+      "Supported OS::macOS"      
+    ]
```
[the remark this answers is not in the document]
Comment (mkarnowski) [reply ngrok/manifest.json]: done

**Comment by mkarnowski** [comment]
> Thanks for submitting your PR! I left some comments and suggested changes. Additionally, please update your setup instructions to include oauth

@emarsha94 our integration doesn't currently support oauth.

File: ngrok/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+ngrok delivers instant ingress to your applications in any cloud, private network, or devices with authentication, load balancing, and other critical controls via our global points of presence.
+
+The ngrok platform includes a Datadog event destination integration.  When used in conjunction with ngrok HTTP events the visualization of valuable application insights is made possible along with the Datadog's advanced log analysis.  Using the the Datadog HTTPS logging endpoint, the integration can quickly be setup through the [ngrok dashboard UI][2].
+
+
+
+## Setup
+
+To forward ngrok events for consumption into Datadog there are two configurations to be made.
```
Comment (alai97) [inline ngrok/README.md:11]:
```suggestion
To forward ngrok events for consumption into Datadog, you need to make two configurations.
```

File: ngrok/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+- ngrok Event Subscription: contains which events to be forwarded
+- ngrok Event Destination: the configuration for where the events defined in the Event Subscription will be forwarded to.
```
Comment (alai97) [inline ngrok/README.md:14]:
```suggestion
- ngrok Event Subscription: Contains which events to be forwarded
- ngrok Event Destination: The configuration for where the events defined in the Event Subscription are forwarded to.
```

File: ngrok/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+Below is an example of configuring an Event Subscription with a Datadog Event Destination for HTTP request events. For step-by-step instructions, see the [ngrok Datadog Event Destination documentation page][7].
```
Comment (alai97) [inline ngrok/README.md:16]:
```suggestion
The following example demonstrates how to configure an Event Subscription with a Datadog Event Destination for HTTP request events. For step-by-step instructions, see the [ngrok Datadog Event Destination documentation][7].
```

File: ngrok/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+**Step 1: Create a ngrok Event Subscription**
+
+1. In the ngrok Dashboard Console, navigate to Events page.
+2. Select "New Subscription".
+3. Provide a Description for Subscription and select "Add Source"
```
Comment (alai97) [inline ngrok/README.md:22]:
```suggestion
3. Provide a Description for Subscription and select "Add Source".
```

File: ngrok/README.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+4. From the list, select "http_request_complete.v0" and select Add Event Source.
+
+
+
+**Step 2: Configure Event Destination properties**
+
+The steps are performed within the previously created Event Subscription configuration:
+
+1. Navigate to the "Event Destination" tab and select "Add Destination".
+2. From the dropdown choose Datadog and input the correct information:\
```
Comment (alai97) [inline ngrok/README.md:32]:
```suggestion
2. From the dropdown menu, select Datadog and input the correct information:
```

File: ngrok/README.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    a. Select the correct Datadog Site for your data - this can be found based upon the [url used to access Datadog][10].\
```
Comment (alai97) [inline ngrok/README.md:33]:
```suggestion
    a. Select the correct [Datadog site][10] for your data.\
```

File: ngrok/README.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    b. Navigate to Datadog and [create an API key][4] within the organization settings.\
+    c. Copy the API key and paste into the API Key field.\
+    d. Optional:  Define a Service Name, this be added as a key to the event data as **service:value**.\
+    e. Optional:  Define DD Tags, these are key:value pairs to be added as Datadog tags to the event data.\
+    f. Optional:  Define a description, this is locally significant and helps identify the Datadog Event Destination.
```
Comment (alai97) [inline ngrok/README.md:38]:
```suggestion
    d. Optionally, define a Service Name, this be added as a key to the event data as **service:value**.\
    e. Optionally, define DD Tags, these are `key:value` pairs to be added as Datadog tags to the event data.\
    f. Optional, define a description, this is locally significant and helps identify the Datadog Event Destination.
```

File: ngrok/README.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+3. Select "Send Test Event"
+4. If presented with a Success message, select "Done".  If an error is presented validate the Datadog Site and API Key are correct.
```
Comment (alai97) [inline ngrok/README.md:40]:
```suggestion
3. Select "Send Test Event".
4. If presented with a Success message, select "Done".  If an error is presented, validate that the Datadog site and API key are correct.
```

File: ngrok/README.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+
+**Step 3: Create Datadog Log Facets**
+Once logs begin to arrive, it will be necessary to create [log facets][8] for data analysis and dashboard visualization. Log facet creation is straight forward and can be accomplished from the Datadog log side panel with guidance available [here][9].
```
Comment (alai97) [inline ngrok/README.md:44]:
```suggestion
Once logs begin to arrive, create [log facets][8] for data analysis and dashboard visualization. For more information about creating log facets, see the [Log Facets documentation][9]. 
```

File: ngrok/README.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+- object.conn.client_ip
+- object.conn.server_name
+- object.http.response.status_code
+- object.http.request.method
+- object.http.request.url.path
+- object.conn.server_name
+
+## Troubleshooting
+
+Need help? Contact [ngrok Support][1] or reference [ngrok Docs][6].
```
Comment (alai97) [inline ngrok/README.md:59]:
```suggestion
Need help? Contact [ngrok Support][1] or reference the [ngrok documentation][6].
```

File: ngrok/README.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+## Further Reading
+
+Learn more about [ngrok][3].
+
+[1]: mailto:support@ngrok.com
+[2]: https://dashboard.ngrok.com
+[3]: https://ngrok.com/solutions
+[4]: https://docs.datadoghq.com/account_management/api-app-keys/
+[6]: https://ngrok.com/docs/integrations/datadog/event-destination/
+[7]: https://ngrok.com/docs/integrations/datadog/event-destination/
```
Comment (alai97) [inline ngrok/README.md:70]:
```suggestion
[6]: https://ngrok.com/docs/integrations/datadog/
[7]: https://ngrok.com/docs/integrations/datadog/event-destination/
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Some copy suggestions!

**Review by alai97 — APPROVED** [review]
Looks good pending Emily's feedback + small merge conflict!

File: ngrok/manifest.json [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        "image_url": "images/diag2.png"
+      }
+    ],
+    "classifier_tags": [
+      "Category::Developer Tools",
+      "Category::Cloud",
+      "Supported OS::Linux",
+      "Supported OS::Windows",
+      "Supported OS::macOS",
+      "Queried Data Type::Logs"
```
Comment (emarsha94) [inline ngrok/manifest.json:36]: Please include a submitted data type if you are sending data to Datadog

File: ngrok/assets/dashboards/ngrok_http_events.json
Comment (emarsha94) [inline ngrok/assets/dashboards/ngrok_http_events.json:1]: Please include a logo with your dashboard

File: ngrok/manifest.json [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+  "manifest_version": "2.0.0",
+  "app_uuid": "3b096ceb-d7a5-4bb5-bf0a-3a07d308d56a",
+  "app_id": "ngrok",
+  "display_on_public_website": true,
+  "tile": {
+    "overview": "README.md#Overview",
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Visualize valuable application insights with ngrok HTTP events.",
```
Comment (emarsha94) [inline ngrok/manifest.json:11]:
```suggestion
    "description": "Visualize valuable application insights with ngrok HTTP events",
```

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: ngrok/manifest.json (hunk 17 above)
Comment (emarsha94) [reply ngrok/manifest.json:36]: As seen [here](https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags)

Comment (mkarnowski) [reply ngrok/manifest.json]: submitted data type added

**Review by yzhan289 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 19]
```diff
@@ -100,6 +100,7 @@
 /neoload/                                @Neotys-Labs/r-d @DataDog/ecosystems-review
 /neutrona/                               @DavidFlamini david@neutrona.com
 /nextcloud/                              @eplanet
+/ngrok/                                  @mkarnowski
```
Comment (emarsha94) [inline .github/CODEOWNERS:103]: Please add @DataDog/ecosystems-review to the code owners file

File: ngrok/CHANGELOG.md [hunk 20]
```diff
@@ -0,0 +1,4 @@
+# CHANGELOG - ngrok
+
+## 1.0.0 / 2023-05-22
```
Comment (emarsha94) [inline ngrok/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2023-06-20
```

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Minor changes but this looks good! Thank you again for all of your hard work :)

File: .github/CODEOWNERS (hunk 19 above)
Comment (mkarnowski) [reply .github/CODEOWNERS]: requested change committed

**Review by emarsha94 — APPROVED** [review]

File: ngrok/assets/service_checks.json
Comment (bgoldberg122) [inline ngrok/assets/service_checks.json:1]: This file can be deleted -- it's only required for app listings that have an `integration` defined within the `asset` object in the `manifest.json`

File: .github/CODEOWNERS [hunk 21]
```diff
@@ -100,6 +100,7 @@
 /neoload/                                @Neotys-Labs/r-d @DataDog/ecosystems-review
 /neutrona/                               @DavidFlamini david@neutrona.com
 /nextcloud/                              @eplanet
+/ngrok/                                  @mkarnowski @DataDog/ecosystems-review
```
Comment (bgoldberg122) [inline .github/CODEOWNERS:103]: Could you also please add entries under `# Community Partners + Documentation` (see `neoload` as an example)?

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 21 above)
Comment (mkarnowski) [reply .github/CODEOWNERS:103]: CODEOWNERS updated

File: .github/CODEOWNERS [hunk 22]
```diff
@@ -390,6 +391,8 @@
 /nextcloud/*metadata.csv                 @eplanet @DataDog/documentation
 /nextcloud/manifest.json                 @eplanet @DataDog/documentation
 /nextcloud/README.md                     @eplanet @DataDog/documentation
+/ngrok/manifest.json                     @mkarnowski @DataDog/documentation
+/ngrok/README.json                       @mkarnowski @DataDog/documentation
```
Comment (bgoldberg122) [inline .github/CODEOWNERS:395]:
```suggestion
/ngrok/*metadata.csv                   @mkarnowski @DataDog/documentation @DataDog/ecosystems-review
/ngrok/manifest.json                     @mkarnowski @DataDog/documentation @DataDog/ecosystems-review
/ngrok/README.json                       @mkarnowski @DataDog/documentation @DataDog/ecosystems-review
```

**Review by bgoldberg122 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 23]
```diff
@@ -390,6 +391,9 @@
 /nextcloud/*metadata.csv                 @eplanet @DataDog/documentation
 /nextcloud/manifest.json                 @eplanet @DataDog/documentation
 /nextcloud/README.md                     @eplanet @DataDog/documentation
+/ngrok/*metadata.csv                   @mkarnowski @DataDog/documentation @DataDog/ecosystems-review
```
Comment (bgoldberg122) [inline .github/CODEOWNERS:394]:
```suggestion
/ngrok/*metadata.csv                     @mkarnowski @DataDog/documentation @DataDog/ecosystems-review
```

**Pull request merged**